### PR TITLE
Fix: load correct article and description for the featured article

### PR DESCRIPTION
**Pull request by cooltey**
This is a temporary fix for T355192

### CODE REVIEW

File: app/src/main/java/org/wikipedia/language/LanguageUtil.kt [hunk 1]
```diff
@@ -144,4 +151,28 @@ object LanguageUtil {
     fun formatLangCodeForButton(languageCode: String): String {
         return languageCode.replace("-", "-\n")
     }
+
+    // TODO: This is a temporary fix for T355192
+    suspend fun getPageSummary(pageSummary: PageSummary?, wikiSite: WikiSite): PageSummary? {
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/language/LanguageUtil.kt:156]: I'm feeling like a `Util` class should contain completely independent bits of code, with as few dependencies as possible. This feels like it's leaking too much stuff into a Util class. Do you think we could keep this as a private function in `AggregatedFeedContentClient`, to discourage its use in other places?

**Review by dbrant — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/wikipedia/language/LanguageUtil.kt (hunk 1 above)
Comment (cooltey) [reply app/src/main/java/org/wikipedia/language/LanguageUtil.kt]: You are right! Done.

**Comment by cooltey** [comment]
Will have a follow-up PR using `prop=extracts` API instead of `page/summary`, because the article content itself loses some text inside the parentheses.

**Comment by dbrant** [comment]
> Will have a follow-up PR using `prop=extracts` API instead of `page/summary`, because the article content itself loses some text inside the parentheses.

Sure, actually I'd like to discuss this further at the offsite, but may I suggest:
The [TextExtracts](https://www.mediawiki.org/wiki/Extension:TextExtracts) extension (which provides `prop=extracts`) is extremely old and unmaintained, and we should probably reduce our dependency on it. If you see issues with the output of the `page/summary` endpoint, it might be better for us to take ownership and update its output so that it meets our needs.

**Comment by cooltey** [comment]
> Sure, actually I'd like to discuss this further at the offsite, but may I suggest:
> The [TextExtracts](https://www.mediawiki.org/wiki/Extension:TextExtracts) extension (which provides `prop=extracts`) is extremely old and unmaintained, and we should probably reduce our dependency on it. If you see issues with the output of the `page/summary` endpoint, it might be better for us to take ownership and update its output so that it meets our needs.

Got it, it makes sense to me. I'll look into the endpoint and try to fix it.

**Pull request merged**